bin/storm.py

```diff
@@ -232,44 +271,91 @@ def jar(jarfile, klass, *args):
     """
+    global DEP_JARS_OPTS, DEP_ARTIFACTS_OPTS
+
+    local_jars = DEP_JARS_OPTS
+    artifact_to_file_jars = resolve_dependencies(DEP_ARTIFACTS_OPTS)
+
     transform_class = confvalue("client.jartransformer.class", [CLUSTER_CONF_DIR])
     if (transform_class != None and transform_class != "nil"):
         tmpjar = os.path.join(tempfile.gettempdir(), uuid.uuid1().hex+".jar")
         exec_storm_class("org.apache.storm.daemon.ClientJarTransformerRunner", args=[transform_class, jarfile, tmpjar], fork=True, daemon=False)
+        extra_jars = [tmpjar, USER_CONF_DIR, STORM_BIN_DIR]
+        extra_jars.extend(local_jars)
+        extra_jars.extend(artifact_to_file_jars.values())
         topology_runner_exit_code = exec_storm_class(
                 klass,
                 jvmtype="-client",
-                extrajars=[tmpjar, USER_CONF_DIR, STORM_BIN_DIR],
+                extrajars=extra_jars,
                 args=args,
                 daemon=False,
                 fork=True,
-                jvmopts=JAR_JVM_OPTS + ["-Dstorm.jar=" + tmpjar])
+                jvmopts=JAR_JVM_OPTS + ["-Dstorm.jar=" + tmpjar] +
+                        ["-Dstorm.dependency.jars=" + ",".join(local_jars)] +
+                        ["-Dstorm.dependency.artifacts=" + json.dumps(artifact_to_file_jars)])
         os.remove(tmpjar)
         sys.exit(topology_runner_exit_code)
     else:
+        extra_jars=[jarfile, USER_CONF_DIR, STORM_BIN_DIR]
+        extra_jars.extend(local_jars)
+        extra_jars.extend(artifact_to_file_jars.values())
         exec_storm_class(
             klass,
             jvmtype="-client",
-            extrajars=[jarfile, USER_CONF_DIR, STORM_BIN_DIR],
+            extrajars=extra_jars,
             args=args,
             daemon=False,
-            jvmopts=JAR_JVM_OPTS + ["-Dstorm.jar=" + jarfile])
+            jvmopts=JAR_JVM_OPTS + ["-Dstorm.jar=" + jarfile] +
+                    ["-Dstorm.dependency.jars=" + ",".join(local_jars)] +
+                    ["-Dstorm.dependency.artifacts=" + json.dumps(artifact_to_file_jars)])
 
 def sql(sql_file, topology_name):
     """Syntax: [storm sql sql-file topology-name]
 
     Compiles the SQL statements into a Trident topology and submits it to Storm.
+
+    --jars and --artifacts options available for jar are also applied to sql command.
+    Please refer "help jar" to see how to use --jars and --artifacts options.
+    You normally want to pass these options since you need to set data source to your sql which is an external storage in many cases.
     """
+    global DEP_JARS_OPTS, DEP_ARTIFACTS_OPTS
+
+    local_jars = DEP_JARS_OPTS
+    artifact_to_file_jars = resolve_dependencies(DEP_ARTIFACTS_OPTS)
+
+    sql_core_jars = get_jars_full(STORM_DIR + "/external/sql/storm-sql-core")
```

Comment:
Definitely. I've also stated that thing to somewhere other pull requests.
`external` has various kinds of modules, and we need to arrange that. 

Maybe we can leave kind of connectors as it is (we can even rename on this), and get flux, sql, kafka-monitor (this module is referred from script in bin, too), and more to out of external.

Since this issue is not only regarding to storm-sql I would like to initiate the discussion before making the change, and handle this to another issue. What do you think?